addon/scheduler.ts

```diff
@@ -60,22 +64,35 @@ export function endTransition(): void {
  * Correct connects the router's transition events to
  * app scheduler's work.
  *
- * @param {Router} router An instance of an Ember router.
+ * @param {RouterService|Router} router An instance of a RouterService or an Ember Router.
  */
-export function setupRouter(router: Router): void {
+export function setupRouter(router: RouterService | Router): void {
   if (IS_FASTBOOT || (router as any)[APP_SCHEDULER_HAS_SETUP]) {
     return;
   }
 
+  deprecate(
+    'The use of Ember.Router in setupRouter is deprecated. Please use the router service.',
+    router instanceof Router,
```

Comment:
@scalvert this deprecation might need to be revisited, see https://github.com/ember-app-scheduler/ember-app-scheduler/issues/956